src/live-programming.rst

```diff
@@ -36,8 +36,132 @@ Direct Manipulation Programming
+  In a high level, the author classified the difficulties into two categories: problems due to overall project size and complexity and problems due to highly domain-specific requirements.
+  Problems due to overall project size and complexity include the lack of excellent development tools, the inefficiency of development workflow, the barriers for multi-platform development, and the lack of excellent third-party components/engines.
+  Problems due to highly domain-specific requirements include the domain-specific knowledge about programming, mathematics, algorithms, computer graphics, etc.
+
+.. container:: bib-item
+
+  .. bibliography:: game-development.bib
+    :filter: key == 'washburn2016went'
+
+  This paper presents an analysis of 155 postmortems published on the gaming site Gamasutra.com. 
+  The paper identifies characteristics of game development, link the characteristics to positive and negative experiences in the postmortems and distill a set of best practices and pitfalls for game development.
+  Based on the analysis, the authors made the following recommendations to game developers: First, be sure to practice good risk management techniques;
+  Second, prescribe to an iterative development process and utilize prototypes as a method of proving features and concepts before committing them to the design;
+  Third, do not be overly ambitious in the design.
+
+As we can see in the above paper, there is a huge problem space in the game development domain including product (e.g. art, creativity, game design), development (e.g. tools, development process, team), resources (budge, hardware, publishers), and customer facing (e.g. marketing, feedback).
+In this compendium, our scope covers only a tiny amount of the whole problem space (i.e. **tools** and **languages** for game development).
+
+Professional Platforms for Game Development
+-------------------------------------------------
+In recent years, game engines such as Unity3D and Unreal Engine have been the most popular platforms for game development.
+A game engine simplifies the task of the programmer by offering convenient abstractions for the hardware and operating systems on top of which the game runs.
+Game engines also provides reusable components such as physics, game inputs, rendering, and Artificial Intelligence (AI).
+Entity-component system and event-handling mechanism have been the major architecture of game engines nowadays.
+
+.. container:: bib-item
+
+  .. bibliography:: game-development.bib
+    :filter: key == 'anderson2008case'
+
+  This article presents a number of potential research topics with regard to the game engine development and architecture design.
+
+.. container:: bib-item
+
+  .. bibliography:: game-development.bib
+    :filter: key == 'messaoudi2015dissecting'
+
+  This paper dissected and analyzed the CPU and GPU performances of one of the most popular game engines, namely Unity3D.
```

Comment:
Do we really want to cover performance issues in game engines in the compendium?